src/codemagic/apple/resources/enums.py

```diff
@@ -214,9 +220,18 @@ def is_tvos_profile(self) -> bool:
         return self.value.startswith('TVOS_')
 
     def devices_not_allowed(self) -> bool:
-        return not self.devices_allowed()
+        return not self.devices_required()
 
     def devices_allowed(self) -> bool:
+        warning = (
+            'Deprecation warning! Method '
+            '"devices_allowed" is deprecated in favor of "devices_required" in version 0.31.3 '
+            'and is subject for removal in future releases.'
+        )
+        log.get_logger(self.__class__).warning(Colors.RED(warning))
```

Comment:
Warning logs should be yellow if you really want to color them, but I don't think this will be necessary. But definitely don't use red, it is for exceptions.